client/modules/IDE/components/Header/Nav.jsx

```diff
@@ -1,414 +1,318 @@
-  user: PropTypes.shape({
-    authenticated: PropTypes.bool.isRequired,
-    username: PropTypes.string,
-    id: PropTypes.string
-  }).isRequired,
-  project: PropTypes.shape({
-    id: PropTypes.string,
-    name: PropTypes.string,
-    owner: PropTypes.shape({
-      id: PropTypes.string,
-      username: PropTypes.string
-    })
-  }),
-  logoutUser: PropTypes.func.isRequired,
-  showShareModal: PropTypes.func.isRequired,
-  showErrorModal: PropTypes.func.isRequired,
-  unsavedChanges: PropTypes.bool.isRequired,
-  showKeyboardShortcutModal: PropTypes.func.isRequired,
-  cmController: PropTypes.shape({
-    tidyCode: PropTypes.func,
-    showFind: PropTypes.func,
-    showReplace: PropTypes.func,
-    getContent: PropTypes.func
-  }),
-  startSketch: PropTypes.func.isRequired,
-  stopSketch: PropTypes.func.isRequired,
-  newFile: PropTypes.func.isRequired,
-  newFolder: PropTypes.func.isRequired,
-  layout: PropTypes.oneOf(['dashboard', 'project']),
-  rootFile: PropTypes.shape({
-    id: PropTypes.string.isRequired
-  }).isRequired,
-  t: PropTypes.func.isRequired,
-  setLanguage: PropTypes.func.isRequired,
-  language: PropTypes.string.isRequired,
-  isUserOwner: PropTypes.bool.isRequired,
-  editorLink: PropTypes.string
+  return (
+    <NavDropdownMenu id="lang" title={languageKeyToLabel(language)}>
+      {sortBy(availableLanguages).map((key) => (
+        // eslint-disable-next-line react/jsx-no-bind
+        <NavMenuItem key={key} value={key} onClick={handleLangSelection}>
+          {languageKeyToLabel(key)}
+        </NavMenuItem>
+      ))}
+    </NavDropdownMenu>
+  );
 };
 
-Nav.defaultProps = {
-  project: {
-    id: undefined,
-    owner: undefined
-  },
-  cmController: {},
-  layout: 'project',
-  editorLink: '/'
+const UnauthenticatedUserMenu = () => {
+  const { t } = useTranslation();
+  return (
+    <ul className="nav__items-right" title="user-menu">
+      {getConfig('TRANSLATIONS_ENABLED') && <LanguageMenu />}
+      <li className="nav__item">
+        <Link to="/login" className="nav__auth-button">
+          <span className="nav__item-header" title="Login">
+            {t('Nav.Login')}
+          </span>
+        </Link>
+      </li>
+      <span className="nav__item-or">{t('Nav.LoginOr')}</span>
+      <li className="nav__item">
+        <Link to="/signup" className="nav__auth-button">
+          <span className="nav__item-header" title="SignUp">
+            {t('Nav.SignUp')}
+          </span>
+        </Link>
+      </li>
+    </ul>
+  );
 };
 
-function mapStateToProps(state) {
-  return {
-    project: state.project,
-    user: state.user,
-    unsavedChanges: state.ide.unsavedChanges,
-    rootFile: state.files.filter((file) => file.name === 'root')[0],
-    language: state.preferences.language,
-    isUserOwner: getIsUserOwner(state),
-    editorLink: selectSketchPath(state)
-  };
-}
+const AuthenticatedUserMenu = () => {
+  const user = useSelector((state) => state.user);
```

Comment:
Seems like we don't need the whole `user` object here either.  Just the `username`.